### PR TITLE
glooctl get proxy: remove --proxy-ns and --all flags

### PR DESCRIPTION
**Pull request by jenshu**
# Description

Remove the `--proxy-ns` and `--all` flags that were added in https://github.com/solo-io/gloo/pull/9226

Previously, Proxies generated by the k8s gateway integration were saved in the same namespace as the Gateway and we didn't have a way to retrieve them with `glooctl get proxy` as it was only checking in the gloo install namespace. As the Proxies are now written to the writeNamespace (discoveryNamespace), we can derive the namespace from Settings instead of having the user pass it in via `--proxy-ns`.

The `--all` flag was useful when proxies could be generated in different namespaces, but now that they are all written to the writeNamespace, we can simplify the CLI by removing that flag.

Other changes: print `proxy namespace` and `created by` in the results table for `glooctl get proxy`

## Testing steps

Manually verified that `glooctl get proxy` can get the proxies in the write namespace, without the proxy-ns flag. (Note: upcoming PRs will add e2e tests and fix the `--edge` / `--kube` filtering for `glooctl get proxy`)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by nfuden — DISMISSED** [review]

**Review by sam-heilbron — DISMISSED** [review]

File: projects/gloo/cli/pkg/common/proxy.go [hunk 1]
```diff
@@ -40,8 +40,13 @@ func GetProxies(name string, opts *options.Options) (gloov1.ProxyList, error) {
 	if err != nil {
 		return nil, err
 	}
+	// get the namespace where Proxies are written. if discovery namespace is empty, defaults to the gloo install namespace
+	writeNamespace := settings.GetDiscoveryNamespace()
```
[the remark this answers is not in the document]
Comment (jenshu) [reply projects/gloo/cli/pkg/common/proxy.go:44]: currently we support an `--all` flag which returns proxies in all namespaces, so we need a way to specify namespace (or lack thereof) in the ProxyEndpointRequest. i think the `--all` may be useful in the case where you have multiple control planes running on the cluster and want to list all proxies across all gloo instances

[the remark this answers is not in the document]
Comment (sam-heilbron) [reply projects/gloo/cli/pkg/common/proxy.go:44]: `i think the --all may be useful in the case where you have multiple control planes running on the cluster and want to list all proxies across all gloo instances` How would that work though? If you have multiple control planes running, glooctl would need to make a separate request to each control plane.

[the remark this answers is not in the document]
Comment (jenshu) [reply projects/gloo/cli/pkg/common/proxy.go:44]: yeah it might only work if persistProxySpec is true and the proxy client is reading all from etcd.

i was considering removing the --all flag altogether, because technically for a given control plane (which you're already required to filter by, via the `-n` flag) it should only return proxies in its write namespace. currently it's possible to have stale proxies in other namespaces (e.g. if you changed the write namespace) and the `--all` would allow the user to see the stale proxies as well (it seems like buggy behavior to have stale proxies at all, so not sure we should expose this via the `--all` flag)

update after some offline discussion:
- we decided to remove the `--all` flag altogether. it's no longer useful, since currently all proxies are written to the write namespace. i'll update this PR to remove the --all flag
- if users still need a way to get proxies across multiple namespaces, they can query the proxy debug endpoint directly

**Review by sam-heilbron — APPROVED** [review]

File: changelog/v1.17.0-beta29/glooctl-remove-proxy-ns.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+changelog:
+  - type: FIX
+    issueLink: https://github.com/solo-io/solo-projects/issues/6088
+    resolvesIssue: false
+    description: >-
+      Removed the `--proxy-ns` flag from `glooctl get proxy`, as proxies are now all written to the discoveryNamespace specified in settings.
```
Comment (npolshakova) [inline changelog/v1.17.0-beta29/glooctl-remove-proxy-ns.yaml:6]: [nit] Do we want to update the changelog to also include removing `--all`

File: projects/gloo/cli/pkg/common/proxy.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (npolshakova) [reply projects/gloo/cli/pkg/common/proxy.go:42]: So when there are multiple control planes installed, what's the result of `glooctl-darwin-arm64 get proxy`? Does it get all proxies across all control planes?

**Review by npolshakova — APPROVED** [review]

File: projects/gloo/cli/pkg/common/proxy.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (npolshakova) [reply projects/gloo/cli/pkg/common/proxy.go:42]: Discussed offline-  if you omit --kube and --edge, it will get everything and you can do `--kube --edge` together

File: changelog/v1.17.0-beta29/glooctl-remove-proxy-ns.yaml (hunk 2 above)
Comment (jenshu) [reply changelog/v1.17.0-beta29/glooctl-remove-proxy-ns.yaml:6]: whoops missed that
